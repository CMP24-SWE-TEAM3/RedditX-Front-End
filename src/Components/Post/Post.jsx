// imports
import React from 'react';
import {PostStyled} from "./Post.styled";

/**
 *  Component that displays a post with the specified input placeholder
 * @returns {React.Component}
 */

const Post = () => {
    const Boom = () => {
        alert('re-direct to Create Post Page');
    }
    return (
        <PostStyled id={'create-post-container'} onClick={Boom} type="post" placeholder="Create Post"/>

    );
};

export default Post;
