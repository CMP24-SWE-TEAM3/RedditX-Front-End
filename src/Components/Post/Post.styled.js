import styled from "styled-components";
import Form from "react-bootstrap/Form";

export const PostStyled = styled(Form.Control)`
  flex-grow: 1;
  height: 38px;
  margin-right: 8px;
  outline: none;
  padding: 0 16px;
  font-size: 14px;
  font-weight: 400;
  line-height: 21px;
  font-family: inherit;
  border-radius: 4px;
  box-sizing: border-box;
  background-color: ${({theme}) => theme.background.post_background};
  color: #333;
  border: 1px solid ${({theme}) => theme.background.post_background};
  display: block;
  box-shadow: 0 0 !important;

  &:hover, &:focus {
    background-color: ${({theme}) => theme.background.post_background};
    border-color: ${({theme}) => theme.color.secondary};
    outline-color: ${({theme}) => theme.color.secondary};
  }

  @media only screen and (max-width: 961px) {
    width: 100%;
  }
`