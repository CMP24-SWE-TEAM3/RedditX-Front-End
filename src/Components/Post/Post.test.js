import {shallow} from 'enzyme';

// import Components
import Post from "./Post";

describe("Post Button Component", () => {
    it("this is a test for Post Button", () => {
        expect(shallow(<Post/>)).toMatchSnapshot();

    });
});