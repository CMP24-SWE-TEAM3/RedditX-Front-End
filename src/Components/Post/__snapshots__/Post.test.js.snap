// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Button Component this is a test for Post Button 1`] = `ShallowWrapper {}`;
